src/wallet/rpc/spend.cpp

```diff
@@ -582,6 +596,17 @@ void FundTransaction(CWallet& wallet, CMutableTransaction& tx, CAmount& fee_out,
         coinControl.fAllowWatchOnly = ParseIncludeWatchonly(NullUniValue, wallet);
     }
 
+    if (g_con_sequentiamode && options.exists("fee_asset")) {
+        std::string strFeeAsset = options["fee_asset"].get_str();
+        CAsset feeAsset = GetAssetFromString(strFeeAsset);
+        if (feeAsset.IsNull()) {
+            throw JSONRPCError(RPC_WALLET_ERROR, strprintf("Unknown label and invalid asset hex for fee: %s", feeAsset.GetHex()));
+        }
+        coinControl.m_fee_asset = feeAsset;
+    } else {
+        coinControl.m_fee_asset = tx.vout[0].nAsset.GetAsset();
```

Comment:
If it is working in single asset fee mode, is the 0 vout the fee vout? I think it is the last one.

Reply:
This code is in `FundTransaction`, so the transaction doesn't need to have an explicit fee output (yet). It gets added in `CreateTransactionInternal` which is called later. See: https://github.com/ElementsProject/elements/blob/master/src/wallet/spend.cpp#L1317-L1325

I actually don't know what the correct behavior is if vout already has a fee output at this point. I'll ask the Liquid dev group.

Reply:
I did some testing and it looks like the fee output gets created if it doesn't already exist and overwritten if it does.

Reply:
Anyway, the logic was definitely wrong because we don't ever want to use `vout[0]` in single asset fee mode. Fixed here: https://github.com/SequentiaSEQ/SEQ-Core-Elements/pull/6/commits/417a071f4a9910095f10139f43810d94718127ab